dotnet-authserver/src/TeacherIdentity.AuthServer/Services/EmailVerification/RateLimitStore.cs

```diff
@@ -0,0 +1,46 @@
+using Microsoft.Extensions.Options;
+using StackExchange.Redis;
+
+namespace TeacherIdentity.AuthServer.Services.EmailVerification;
+
+public class RateLimitStore : IRateLimitStore
+{
+    private readonly IConnectionMultiplexer _connectionMultiplexer;
+    private readonly IOptions<RateLimitStoreOptions> _rateLimitOptions;
+    public RateLimitStore(IConnectionMultiplexer connectionMultiplexer, IOptions<RateLimitStoreOptions> rateLimitOptions)
+    {
+        _connectionMultiplexer = connectionMultiplexer ?? throw new ArgumentException("IConnectionMultiplexer was null. Ensure StackExchange.Redis was successfully registered");
+        _rateLimitOptions = rateLimitOptions;
+    }
+
+    public async Task AddFailedPinVerification(string clientIp)
+    {
+        if (await _connectionMultiplexer.GetDatabase().KeyExistsAsync(new RedisKey(clientIp)))
```

Comment:
I think this has a data race; probably need something like is used here: https://github.com/stefanprodan/AspNetCoreRateLimit/blob/master/src/AspNetCoreRateLimit.Redis/RedisProcessingStrategy.cs